### PR TITLE
Add option for matching a file to be transformed

### PR DESCRIPTION
**Pull request by Fattimo**

### CODE REVIEW

**Comment by Fattimo** [comment]
resolves #2

File: src/index.ts [hunk 1]
```diff
@@ -156,7 +157,7 @@ function hijackCSSPostPlugin(
       // result of vite:post
       const result = (await _transform.apply(this, args as any)) as TransformResult
       // this result will be modified if the conditions of vite:css-export are met.
-      if (isCSSRequest(id) && exportRE?.test(id)) {
+      if (isCSSRequest(id) && shouldTransform ? (shouldTransform(id) : exportRE?.test(id))) {
```
Comment (shixuanhong) [inline src/index.ts]: This expression doesn't seem to work properly

File: src/index.ts [hunk 2]
```diff
@@ -192,13 +193,13 @@ export default function ViteCSSExportPlugin(
...
       cssPostPlugin &&
-        hijackCSSPostPlugin(cssPostPlugin, config, cssModule, parseResultCache)
+        hijackCSSPostPlugin(cssPostPlugin, config, cssModule, parseResultCache, options.shouldTransform)
     },
     buildStart() {
       parseResultCache.clear()
     },
     async transform(code, id, options) {
-      if (isCSSRequest(id) && exportRE.test(id)) {
+      if (isCSSRequest(id) && options.shouldTransform ? (options.shouldTransform(id) : exportRE?.test(id))) {
```
Comment (shixuanhong) [inline src/index.ts]: same as that

**Review by shixuanhong — CHANGES_REQUESTED** [review]

**Comment by Fattimo** [comment]
sorry about that! I was not at my usual laptop and thought using the github editor would be simple enough.

I've gone ahead and made the code actually functional 😅 and updated the documentation + example. Let me know if there are any other blockers for this feature

**Comment by Fattimo** [comment]
image
image

File: src/index.ts [hunk 3]
```diff
@@ -156,7 +157,7 @@ function hijackCSSPostPlugin(
       // result of vite:post
       const result = (await _transform.apply(this, args as any)) as TransformResult
       // this result will be modified if the conditions of vite:css-export are met.
-      if (isCSSRequest(id) && exportRE?.test(id)) {
+      if (isCSSRequest(id) && shouldTransform ? shouldTransform(id) : exportRE?.test(id)) {
```
Comment (shixuanhong) [inline src/index.ts:160]: Maybe you don't understand what I mean, I prefer both to work, not just one or the other. (#2 )

Like  `shouldTransform(id) || exportRE.test(id) `

File: client.d.ts [hunk 4]
```diff
@@ -31,3 +31,11 @@ declare module '*.stylus?export' {
   const cssExportVariables: CSSPropertiesExportedData
   export default cssExportVariables
 }
+
+/**
+ * If `shouldTransform` is used, you must handle type checking for special cases like this yourself.
+ */
+declare module '*.scss?shouldTransformString' {
```
Comment (shixuanhong) [inline client.d.ts:38]: If you are just writing an example, please do not modify this file, it may be referenced by users.

**Pull request closed without merging**